File: mentorconnect/app/schemas/rating.py
"""
app/schemas/rating.py
------------------------
Schemas for students rating mentors after a completed booking.
"""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field


class RatingCreateRequest(BaseModel):
    booking_id: int
    score: int = Field(..., ge=1, le=5)
    review: Optional[str] = Field(None, max_length=2000)


class RatingOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    booking_id: int
    student_id: int
    mentor_id: int
    score: int
    review: Optional[str] = None
    created_at: datetime

File: mentorconnect-frontend/src/types/rating.ts
export interface Rating {
  id: number;
  booking_id: number;
  student_id: number;
  mentor_id: number;
  score: number;
  review?: string | null;
  created_at: string;
}

export interface CreateRatingRequest {
  booking_id: number;
  score: number;
  review?: string;
}